lib/grpnames.gi

```diff
@@ -20,46 +20,575 @@
 ##  from GAP3 code written by Markus Püschel and Sebastian Egner.
 ##
 
+
+#############################################################################
+##
+#M  IsTrivialNormalIntersectionInList( <L>, <U>, <V> ) . . . . generic method
+##
+InstallMethod( IsTrivialNormalIntersectionInList,
+               "if minimal normal subgroups are computed",
+               [ IsList, IsGroup, IsGroup ],
+
+  function( L, U, V )
+    local gs;
+
+    gs:= List(Filtered(L, N -> not IsTrivial(N)),
+                                               N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
+  end);
```

Comment:
Note that using List + Filtered + ForAll here means that you are allocating a lot of temporary objects and memory, always -- and even if the very first check in the ForAll returns "false".
So, this could be optimized by using loops instead, which would reduce both memory and time consumption. Something like this (untested, I just typed it in here, and I might have gotten the logic wrong, too):
```gap
function( L, U, V)
  local N, g;
  for N in L do
    if IsTrivial(N) then continue; fi;
    g := GeneratorsOfGroup(N)[1];
    if g in U and g in V then return false; fi;
  od;
  return true;
end
```